### DISCUSSION TITLE
How to exclude InfoBarMessageControl in Orchestra.Shell.MahApps?

### DISCUSSION THREAD

**Discussion by NickHeinz-Andretti** [WildGums/Orchestra#542] [Q&A]
Building an application using the MahApps shell, but would like to exclude the InfoBarMessageControl.  It seems the option is in the Catel DataWindow constructor to set the generation property, but it doesn't appear this option is exposed in the ShellWindow constructor.  What would be the proper method for disabling (InfoBarMessageControlGenerationMode.None) while using the MahApps shell?

Thanks!

**Comment by GeertvanHorrik** [comment]
Good question, I think it's not possible out of the box. For now a workaround would be to use private reflection to set this field using this pseudo code:

```
var field = GetType().GetFieldEx("_infoBarMessageControlGenerationMode");
field.SetValue(this, InfoBarMessageControlGenerationMode.None);
```

Make sure to call this code *before* `OnInitialized` is called the WPF.

**Reply by NickHeinz-Andretti** [reply]
Thanks for the reply - good idea, I didn't think of using reflection to set it.

I'm struggling with the "entry point" to change the field using the above though.  It seems that once `shellService.CreateAsync<ShellWindow>()` is called inside of App.xaml.cs, I don't have a way to access the `shellService.Shell` property and change the field before it's initialized.  Using `ApplicationInitializationService.InitializeBeforeCreatingShellAsync()` is too early (no Shell), and `InitializeAftercreatingShellAsync()` is too late (IsInitialized has already been set).  Am I missing an obvious spot to modify it?

I've thought of trying to 'extend' the `ShellService.OnCreatingShell()` method, but the compiler doesn't like that my extension is in a different assembly.

**Reply by GeertvanHorrik** [reply]
I think `Application.Windows.LastOrDefault()` should give you the shell. While the splash screen is up, I think you cannot use `Application.MainWindow` since that will probably be the splash screen.

**Reply by NickHeinz-Andretti** [reply]
Ok thanks, I'll give that a shot.

**Reply by NickHeinz-Andretti** [reply]
`Application.Windows.LastOrDefault()` can give me the shell, but I'm still stuck on when to make the modification.  If I wait until the the `ApplicationInitializationService.InitializeAftercreatingShellAsync()` it's too late.  Any time before the ShellService calls `CreateShellAsync()` (specifically the line [here](https://github.com/WildGums/Orchestra/blob/668c64920f364ca4e01008c07147db5a53c66504/src/Orchestra.Shell.Shared/Services/ShellService.cs#L291) I believe) then it's too early and I can't change the field.  I'm still unsure where to add this in.

**Reply by NickHeinz-Andretti** [reply]
Hi @GeertvanHorrik, wondering if you had any suggestions for where to "enter" and change the field before the `OnInitialized` is called.  Thanks!
